Chatter/AvatarPicker/AvatarPickerRootView.swift

```diff
@@ -0,0 +1,135 @@
+//
+//  AvatarPickerRootView.swift
+//  Chatter
+//
+
+import Foundation
+
+public class AvatarPickerRootView: UIView {
+
+    typealias AvatarPickerDismissHandler = () -> Void
+    typealias AvatarTypeChangeHandler = (AvatarType) -> Void
+
+    var delegate: UICollectionViewDelegate? {
+        set {
+            collectionView.delegate = newValue
+        }
+        get {
+            return nil
+        }
+    }
+    var dataSource: UICollectionViewDataSource? {
+        set {
+            collectionView.dataSource = newValue
+        }
+        get {
+            return nil
+        }
+    }
+
+    var dismissHandler: AvatarPickerDismissHandler?
+    var avatarTypeChangeHandler: AvatarTypeChangeHandler?
+
+    private let backButton: UIButton = {
+        let backButton = UIButton(type: .custom)
+        backButton.setImage(#imageLiteral(resourceName: "smackBack"), for: .normal)
+        backButton.heightAnchor
+            .constraint(equalToConstant: 30)
+            .isActive = true
+        backButton.widthAnchor
```

Comment:
Consider moving those constraints into the `activateConstraintsBackButton` function to keep all constraints in one place.